vep/run_vep.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script is intended to be used in the Docker image built for VEP.
+# Note that the arguments are passed through environment variables and to log
+# useful error messages when they are not properly set, we always check their
+# existence the first time they are being accessed.
+#
+# The only enviroment variables that have to be set are (others are optional):
+#
+# VEP_CACHE: The path of the VEP cache which is a single .tar.gz file.
+# INPUT_FILE: The path of the input file (might be a VCF or a compressed VCF).
+# OUTPUT_VCF: The path of the output file which is always a VCF file.
+#
+# For the full list of supported environment variables and their documentation
+# check README.md.
+# Capital letter variables refer to environment variables that can be set from
+# outside. Internal variables have small letters.
+
+set -euo pipefail
+
+readonly species="${SPECIES:-homo_sapiens}"
+readonly assembly="${GENOME_ASSEMBLY:-GRCh38}"
+readonly fork_opt="--fork ${NUM_FORKS:-1}"
+readonly other_vep_opts="${OTHER_VEP_OPTS:---everything}"
+readonly annotation_field_name="${VCF_INFO_FILED:-CSQ_VT}"
+
+if [[ ! -r "${VEP_CACHE:?VEP_CACHE is not set!}" ]]; then
+  echo "ERRPR: Cannot read ${VEP_CACHE}"
+  exit 1
+fi
+
+# Check INPUT_FILE is set and is a readable file.
+if [[ ! -r "${INPUT_FILE:?INPUT_FILE is not set!}" ]]; then
+  echo "ERRPR: Cannot read ${INPUT_FILE}"
+  exit 1
+fi
+
+# TODO(bashir2): Add support for multiple input and output files.
+
+echo "Checking input file at $(date)"
+ls -l "${INPUT_FILE}"
+
+# Make sure output file does not exist and can be written.
+if [[ -e ${OUTPUT_VCF:?OUTPUT_VCF is not set!} ]]; then
+  echo "ERROR: ${OUTPUT_VCF} already exits!"
```

Comment:
This is also inside the docker container. So currently, it should never happen (as we always have a fresh container for each input file so there is no output file). This is mainly to prevent obvious errors in the job description config. Once I add support for multiple input files these kind of checks becomes more relevant. Or is there a re-run scenario that I did not realize from your comment?

Reply:
In case of pipelines re-run, there is no problem as a new VM/Container is launched in retry. However, if someone just want to use your script manually inside a VM/Container, then it might be more convenient to just allow the re-write.

Reply:
The drawback of allowing rewrite is that people may accidentally overwrite a big output file that is generated (or being generated) after hours of processing. I think VEP itself, by default, do not overwrite existing files as well. I prefer not to expose the possibility of (unintentional) overwrite.